week2/CSS - Media query with Styled-Components.md

```diff
@@ -0,0 +1,217 @@
+# 반응형 웹 - Media query with Styled-Components
+
+```
+yarn add styled-components
+```
+
+## 1. Mobile First
+
+```jsx
+import styled from 'styled-components';
+
+const Col = styled.div`
+  float: left;
+  width: 100%;
+  padding: 1rem;
+
+  @media only screen and (min-width: 768px) {
+    width: ${(props) => (props.span ? (props.span / 12) * 100 : '8.33')}%;
+  }
+`;
+```
+
+## 2. Breakpoint 추가
+
+- 768px 이하: xs를 span으로 적용
+- 992px 이하: sm을 span으로 적용
+- 1200px 이하: md를 span으로 적용
+- 1200px 이상: lg를 span으로 적용
+
+```jsx
+//Column.js
+const calcWidthPercent = (span) => {
```

Comment:
One of the reasons why we want to recalcluate the number of columns per row is to ensure the min-width. We can probably also use a grid template then. Nice topic!